section3/reflection.md

```diff
@@ -2,16 +2,40 @@
 
 1. How different did your workflow feel this week, considering we asked you to follow the Pomodoro technique?
 
+    - Workflow felt good to step away and then come back. Looking forward to practicing this through Turing. Normally, I can get lost in my work and bang my head around things.
+
 2. Regarding the work you did around setting intentions in Step 1 of the Pomodoro technique - how did that go? Were you surprised by anything (did you find yourself way more focused than you realized, more distracted that you thought you'd be, estimating times accurately or totally off, etc)?
 
+    - I noticed that sometimes that my estimation of completing things does not always align. There were times where things went quicker than expected and then there were others where I really struggled and wanted to learn more before moving on to the next task.
+
+
 3. What is a Hash, and how is it different from an Array?
 
+    - A Hash is a collection of unique keys and their values. It is different from an Array that the indexing is done from the keys and not the order that is in an Array.
+
 4. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
 
+```
+pet_store = {dog food: 234, cat food: 136, dog collars: 26, fish tanks: 35}
+```
+
 5. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
 
+```
+puts states["IA"]
+```
+
 6. With the same hash above, how would we get all the keys?  How about all the values?
 
+```
+puts states.keys
+puts states.values
+```
+
 7. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
 
-8. What questions do you still have about hashes?
\ No newline at end of file
+    - Another example would be for grades. You can have the key be the student and the value be the grade. A hash is better than an array here because instead of just have a list of all the grades you can see which grade is assigned to which student.
+
+8. What questions do you still have about hashes?
+
+    - I feel like I am starting to understand the concepts more about hashes but I think I just need more time and practice on how you can use arrays and hashes together and getting more comfortable on when you should choose an array compared to a hash for a data set. It would be helpful to know if there is a like a list of questions to ask yourself at the beginning to know which type you should be using.
```

Comment:
YES that is a great question. When I taught Mod 1 the question I found myself asking/coaching students on MOST was "what data type are you working with?" because they hadn't slowed down to identify that, then made a mistake as a result.

I would say when looking at array v hash - it's the bracket. Do you see a `[` or `{` - is your first sign.